modules/tinymce/src/plugins/quickbars/test/ts/browser/PredicateTest.ts

```diff
@@ -0,0 +1,51 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { TinyHooks, TinySelections } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/quickbars/Plugin';
+
+import { pAssertToolbarNotVisible } from '../module/test/Utils';
+
+describe('browser.tinymce.plugins.quickbars.PredicateTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    plugins: 'quickbars link',
+    inline: true,
+    toolbar: false,
+    menubar: false,
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin ], true);
+
+
+  it('TINY-9190: Toolbar is not shown when the ancestor element has a data-mce-bogus attribute', async () => {
```

Comment:
You'll probably need to add some tests here for when it should be shown. Otherwise, all of this passes with Fun.constant(false).